File: src/C#/BehavioralPatterns/State/VendingMachine/SoldOutState.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace VendingMachine
{
    public class SoldOutState : State
    {
        public SoldOutState(VendingMachine vendingMachine) : base(vendingMachine)
        {
            Console.WriteLine("SOLDOUT");
        }

        public override void InsertMoney(decimal amount) => Console.WriteLine("There are no products in the vending machine.");

        public override void SelectProduct(string productCode) => Console.WriteLine("There are no products in the vending machine.");

        public override void DispenseProduct() => Console.WriteLine("There is no selected product for dispensing");

        public override void Cancel() => Console.WriteLine("There is no operation to be cancelled");

        public override void Refill(List<Product> products)
        {
            vendingMachine.Products = products;
            Console.WriteLine($"Total amount of products: {vendingMachine.Products.Sum(p => p.Stock)}");
            vendingMachine.SetState(new IdleState(vendingMachine));
        }
    }
}

File: src/JavaScript/BehavioralPatterns/State/VendingMachine/soldOutState.js

const State = require("./state");
const IdleState = require("./idleState");

class SoldOutState extends State {
    constructor(vendingMachine) {
        super(vendingMachine);
        console.log("SOLDOUT");
    }

    insertMoney(amount) {
        console.log("There are no products in the vending machine.");
    }

    selectProduct(productCode) {
        console.log("There are no products in the vending machine.");
    }

    dispenseProduct() {
        console.log("There is no selected product for dispensing");
    }

    cancel() {
        console.log("There is no operation to be cancelled");
    }

    refill(products) {
        this.vendingMachine.products = products;
        console.log(`Total amount of products: ${this.vendingMachine.products.reduce((acc, curr) => acc + curr.stock, 0)}`);
        this.vendingMachine.setState(new IdleState(this.vendingMachine));
    }
}

module.exports = SoldOutState;